test/e2e/framework/nodes.go

```diff
@@ -50,8 +50,9 @@ func FindGatewayNodes(cluster ClusterIndex) []v1.Node {
 func FindNonGatewayNodes(cluster ClusterIndex) []v1.Node {
 	nodes, err := KubeClients[cluster].CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
 		LabelSelector: labels.NewSelector().Add(
-			NewRequirement(GatewayLabel, selection.Exists, []string{}),
-			NewRequirement(GatewayLabel, selection.NotEquals, []string{"true"})).String(),
+			// Ignore the control plane node labeled as master as it doesn't allow scheduling of pods
+			NewRequirement("node-role.kubernetes.io/master", selection.DoesNotExist, []string{}),
+			NewRequirement(GatewayLabel, selection.DoesNotExist, []string{})).String(),
```

Comment:
Since our semantics elsewhere are such that a node is a GW if the gateway label exists **and** is equal to "true", I think we should be consistent here as well. The following handles non-existent label as well (in my testing):

```suggestion
			NewRequirement(GatewayLabel, selection.NotEquals, []string{"true"})).String(),
```

Reply:
Done